LottoDiary/LottoDiary/Sources/Present/Calendar/ViewController/LottoCalendarViewController.swift

```diff
@@ -0,0 +1,410 @@
+    
+    enum Section: Equatable {
+        case date
+    }
+    
+    var dataSource: DataSource!
+    var lottosCollectionView: UICollectionView! = nil
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .designSystem(.gray17181D)
+        
+        // 스크롤 뷰 레이아웃 설정
+        configureScrollView()
+        
+        // 캘린더 레이아웃 설정
+        configureCalendarLayout()
+        // 캘린더 이벤트 등록
+        setupEvents()
+        
+        // CollectionView, DataSource 설정
+        configureCollectionView()
+        configureDataSource()
+        
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        changeHeaderTitle()
+    }
+    
+
+}
+
+
+// MARK: - DiffableDataSource, CompositionalLayout 관련 함수
+extension LottoCalendarViewController {
+    func configureCollectionView() {
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: generateLayout())
+        
+        contentView.addSubview(collectionView)
+        collectionView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        collectionView.delegate = self
+        collectionView.register(LottoCell.self, forCellWithReuseIdentifier: LottoCell.reuseIdentifier)
+        collectionView.backgroundColor = .designSystem(.gray17181D)
+
+        collectionView.isScrollEnabled = false
+        lottosCollectionView = collectionView
+        
+        viewModel.filteredLottos
+            .map { $0.count }
+            .observe(on: MainScheduler.instance)
+            .subscribe(onNext: { count in
+                self.lottosCollectionView.snp.makeConstraints { make in
+                    make.leading.equalToSuperview()
+                    make.top.equalTo(self.calendarHeight + 20) // 캘린더와의 거리
+                    make.width.equalToSuperview()
+                    // 동적으로 데이터가 들어가야함
+                    make.height.equalTo(count * 90 + 20 + 50) // 셀 크기, 헤더(MM월 dd일), 푸터
+                    // bottom으로 contentView를 마무리지음
+                    make.bottom.equalToSuperview()
+                }
+            })
+            .dispose()
+    }
+    
+    func configureDataSource() {
+        dataSource = DataSource(collectionView: lottosCollectionView) {
+            (collectionView: UICollectionView, indexPath: IndexPath, lottoId: Lotto.ID) ->
+            UICollectionViewCell? in
+            
+            guard let cell = collectionView.dequeueReusableCell(
+                withReuseIdentifier: LottoCell.reuseIdentifier,
+                for: indexPath) as? LottoCell
+            else { fatalError("Could not create new cell") }
+            let lotto = self.lotto(for: lottoId)
+            
+            if lotto != nil {
+                cell.purchaseAmount.text = "\(lotto!.purchaseAmount.formattedWithSeparator) 원"
+                cell.winningAmount.text = "\(lotto!.winningAmount.formattedWithSeparator) 원"
+                cell.type = lotto!.type
+            }
+            return cell
+        }
```

Comment:
DataSource = UICollectionViewDiffableDataSource<Section, Lotto> 로 설정하고,
```suggestion
    func configureDataSource() {
            dataSource = DataSource(collectionView: lottosCollectionView) {
            (collectionView, indexPath, identifier) -> UICollectionViewCell? in
            
            guard let cell = collectionView.dequeueReusableCell(
                withReuseIdentifier: LottoCell.reuseIdentifier,
                for: indexPath) as? LottoCell
            else { fatalError("Could not create new cell") }
            
            cell.purchaseAmount.text = "\(identifier.purchaseAmount.formattedWithSeparator) 원"
            cell.winningAmount.text = "\(identifier.winningAmount.formattedWithSeparator) 원"
            cell.type = identifier.type
            return cell
        }
```
이런식으로 구현하면, id를 받아 데이터 중 id가 동일한 Lotto 객체를 만들 필요 없이 cell과 연결할 수 있을 것 같습니다!
(아니면, dataSource에 Lotto 대신 Lotto.ID를 넣은 이유가 있을까요?)

Reply:
Lotto를 hashable한 값으로 사용한다면 내부 id프로퍼티의 효용성이 사라진다고 생각했기 때문에 ID를 넣어줘서 사용했습니다 !

Reply:
오오,,저도 이렇게 바꿔야겠습니다,,